### PR TITLE
vmware.vmware_rest: Generate action_groups

### LINKED ISSUE #75: Generate action_groups in runtime.yml
##### SUMMARY
I think it would be really useful if you create `action_groups` in `meta/runtime.yml` for all the modules generated.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
plugins/action/generate_cloud_modules.py

##### ADDITIONAL INFORMATION
ansible-collections/vmware.vmware_rest#361
ansible-collections/vmware.vmware_rest#440

### PR DESCRIPTION
**Pull request by mariolenz**
##### SUMMARY
Fixes #75

Generate action_groups for `vmware.vmware_rest`.

_edit:_ Tested in ansible-collections/vmware.vmware_rest#461

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/action/generate_cloud_modules.py

##### ADDITIONAL INFORMATION
Follow-up to #76 which I've closed accidentally and can't reopen again.

### CODE REVIEW

**Comment by mariolenz** [comment]
@GomathiselviS Could you please review this? I've refactored the code a bit because it wasn't really feasible for a unit test... I mean, it wasn't a self-contained unit but part of a larger block of code. I hope it's OK now.

cc @alinabuzachis @NilashishC

**Review by alinabuzachis — APPROVED** [review]

File: plugins/action/generate_cloud_modules.py [hunk 1]
```diff
@@ -1325,6 +1327,14 @@ def generate_vmware_rest(args: Iterable, role_path: str):
                     role_path=role_path
                 )
                 module_list.append(module.name)
+
```
Comment (GomathiselviS) [inline plugins/action/generate_cloud_modules.py:1330]:
```suggestion
print("Generating meta/runtime.yml")
```

**Review by GomathiselviS — APPROVED** [review]
Looks good to me. I've recommended adding a log message for clarity.

**Comment by mariolenz** [comment]
> Looks good to me. I've recommended adding a log message for clarity.

[done](https://github.com/ansible-community/ansible.content_builder/pull/78/commits/11064d5b825c8b719ff61ccdba510359d721be98)

**Comment by mariolenz** [comment]
I think the aws collection could make use of this, too. Maybe changing:

https://github.com/ansible-community/ansible.content_builder/blob/e775adc466a906c994d5af3e4c14aac6233b422e/plugins/action/generate_cloud_modules.py#L1238-L1244

to something like:

```
yaml_dict = generate_runtime_yml("2.11.0", "aws", module_list)
yaml_dict["plugin_routing"] = {}
```

or:

```
yaml_dict = generate_runtime_yml(args.get("requires_ansible"), "aws", module_list)
yaml_dict["plugin_routing"] = {}
```

The latter would mean that `requires_ansible` should be also valid for aws, though.

BTW: This is jus a suggestion from my side. I won't implement this, I don't have anything to do with this collection :smirk:

**Comment by Yannik** [comment]
@mariolenz Great, thanks!

**Pull request merged**